internal/providers/terraform/aws/apigatewayv2_api.go

```diff
@@ -0,0 +1,217 @@
+package aws
+
+import (
+	"github.com/infracost/infracost/internal/schema"
+	"github.com/shopspring/decimal"
+)
+
+func GetApiGatewayv2ApiRegistryItem() *schema.RegistryItem {
+	return &schema.RegistryItem{
+		Name:  "aws_apigatewayv2_api",
+		Notes: []string{
+			"WebSocket Connection minutes is not yet supported",
+		},
+		RFunc: NewApiGatewayv2Api,
+	}
+}
+
+func NewApiGatewayv2Api(d *schema.ResourceData, u *schema.ResourceData) *schema.Resource {
+	var costComponents []*schema.CostComponent
+
+	protocolType := d.Get("protocol_type").String()
+
+	if protocolType == "WEBSOCKET" {
+		costComponents = websocketApiCostComponent(d, u)
+	}
+
+	if protocolType == "HTTP" {
+		costComponents = httpApiCostComponent(d, u)
+	}
+
+	return &schema.Resource{
+		Name:           d.Address,
+		CostComponents: costComponents,
+	}
+}
+
+func httpApiCostComponent(d *schema.ResourceData, u *schema.ResourceData) []*schema.CostComponent {
+	region := d.Get("region").String()
+	monthlyRequests := decimal.Zero
+	requestSize := decimal.NewFromInt(512)
+
+	billableRequestSize := decimal.NewFromInt(512)
+
+	var apiTierRequests = map[string]decimal.Decimal{
+		"apiRequestTierOne": decimal.Zero,
+		"apiRequestTierTwo": decimal.Zero,
+	}
+
+	// httpApi request tiers
+	apiTierOneLimit := decimal.NewFromInt(300000000)
+	apiTierTwoLimit := decimal.NewFromInt(300000001)
+
+	if u != nil && u.Get("monthly_requests.0.value").Exists() {
+		monthlyRequests = decimal.NewFromInt(u.Get("monthly_requests.0.value").Int())
+	}
+
+	if u != nil && u.Get("average_request_size.0.value").Exists() {
+		requestSize = decimal.NewFromInt(u.Get("average_request_size.0.value").Int())
+	}
+
+	if requestSize.GreaterThan(billableRequestSize) {
+		monthlyRequests = calculateBillableRequests(requestSize, billableRequestSize, monthlyRequests)
+	}
+
+	apiRequestQuantities := calculateApiV2Requests(monthlyRequests, apiTierRequests, apiTierOneLimit, apiTierTwoLimit)
+
+	apiTierOne := apiRequestQuantities["apiRequestTierOne"]
+	apiTierTwo := apiRequestQuantities["apiRequestTierTwo"]
+
+	CostComponent := []*schema.CostComponent{
+		{
+			Name:            "Requests (first 300m)",
```

Comment:
```suggestion
			Name:            "Requests (first 300M)",
```

I'll add this to the CONTRIBUTING.md file
```
- tiers in names: use the K postfix for thousand, M for million, B for billion and T for trillion, e.g. "Requests (first 300M)" and "Messages (first 1B)". Note the capital case letters.
```